modules/transformation.py

```diff
@@ -0,0 +1,627 @@
+            #   not an issue for the java SDK
+            #     https://beam.apache.org/releases/javadoc/2.37.0/org/apache/beam/sdk/io/contextualtextio/ContextualTextIO.Read.html#withHintMatchesManyFiles--
+            files = (
+                p
+                # TODO replace with match continuously for streaming?
+                | fileio.MatchFiles(matching)
+                | "ExtractPath" >> beam.Map(lambda f: f.path)
+                | beam.Reshuffle()
+                | beam.io.ReadAllFromText(with_filename=True)
+            )
+
+        return files
+
+
+class WriteBigQuery(beam.PTransform):
+    def __init__(self, table, schema, streaming=None):
+        super().__init__()
+        self.table = table
+        self.schema = schema
+        self.streaming = streaming
+
+    def resolve_params(self):
+        if self.streaming:
+            # streaming pipeline
+            return {
+                "create_disposition": BigQueryDisposition.CREATE_IF_NEEDED,
+                "write_disposition": BigQueryDisposition.WRITE_APPEND,
+                "with_auto_sharding": True,
+                # parameters for STREAMING_INSERTS
+                "method": WriteToBigQuery.Method.STREAMING_INSERTS,
+                "ignore_unknown_columns": True,
+                "insert_retry_strategy": RetryStrategy.RETRY_ON_TRANSIENT_ERROR,
+                # parameters for FILE_LOADS
+                # "method": WriteToBigQuery.Method.FILE_LOADS,
+                # "triggering_frequency": 5 * 60,  # seconds
+                # "additional_bq_parameters": {"ignoreUnknownValues": True},
+            }
+        else:
+            # batch pipeline
+            return {
+                "create_disposition": BigQueryDisposition.CREATE_IF_NEEDED,
+                "write_disposition": BigQueryDisposition.WRITE_TRUNCATE,
+                "method": WriteToBigQuery.Method.FILE_LOADS,
+                "temp_file_format": FileFormat.JSON,
+                "additional_bq_parameters": {"ignoreUnknownValues": True},
+            }
+
+    def expand(self, pcoll, **kwargs):
+        return pcoll | WriteToBigQuery(
+            table=self.table, schema=self.schema, **self.resolve_params()
+        )
+
+
+def initialize_status_info(file_name, page):
+    # file name parsing kept for backward compatibility before 2022-03-01
+    dir_name, base_name = os.path.split(file_name)
+
+    date = utils.crawl_date(dir_name)
+
+    metadata = page.get("_metadata", {})
+
+    return {
+        "archive": "All",  # only one value found when porting logic from PHP
+        "label": "{dt:%b} {dt.day} {dt.year}".format(dt=date),
+        "crawlid": metadata.get("crawlid", 0),
+        "wptid": page.get("testID", base_name.split(".")[0]),
+        "medianRun": 1,  # only available in RAW json (median.firstview.run), not HAR json
+        "page": metadata.get(
+            "page", None  # TODO future improvement, not populated as of 2022-05
+        ),
```

Comment:
Extract `_metadata.tested_url` to be used for the `URL` field

```suggestion
        "URL": metadata.get(
            "tested_url", None  # TODO future improvement, not populated as of 2022-05
        ),
        "page": metadata.get(
            "page", None  # TODO future improvement, not populated as of 2022-05
        ),
```